### PR TITLE
tracking of dataset recommendations

### PR DESCRIPTION
**Pull request by taniki**

### CODE REVIEW

File: theme/js/dataset-recommendations.js [hunk 1]
```diff
@@ -20,6 +26,10 @@ function addRecos(recos) {
         recoChildContainer.appendChild(recoChildEmbed);
         recoContainer.appendChild(recoChildContainer);
     });
+
+    if (!(typeof _paq == "undefined")) {
```
Comment (abulte) [inline theme/js/dataset-recommendations.js]: single quotes 😀

File: theme/js/dataset-recommendations.js [hunk 2]
```diff
@@ -20,6 +26,10 @@ function addRecos(recos) {
         recoChildContainer.appendChild(recoChildEmbed);
         recoContainer.appendChild(recoChildContainer);
     });
+
+    if (!(typeof _paq == 'undefined')) {
```
Comment (abulte) [inline theme/js/dataset-recommendations.js]: Proposal, not sure if better (I don't like this syntax).

```
const paq = _paq || {};
if (paq.push) { paq.push [...] }
```

Comment (noirbizarre) [inline theme/js/dataset-recommendations.js]: Why not `if (_paq !== undefined)` or `if (_paq)` ?

Comment (noirbizarre) [reply theme/js/dataset-recommendations.js]: :joy: 
Just published mine below. I don't like this too

Comment (abulte) [reply theme/js/dataset-recommendations.js]: `if (_paq)` will crash. `if (_paq !== undefined)`, I don't remember why but I think it's bad practice vs `typeof`.

Comment (abulte) [reply theme/js/dataset-recommendations.js]: `if (_paq !== undefined)` will crash too actually. But my original proposal would too, I fixed it.

Comment (taniki) [reply theme/js/dataset-recommendations.js]: @abulte your proposal was a bit buggy for some obscure reason of direction of execution.

File: theme/js/dataset-recommendations.js [hunk 3]
```diff
@@ -8,10 +8,19 @@ function getDatasetId() {
 }
 
 function addRecos(recos) {
+    const paq = _paq || {};
```
Comment (noirbizarre) [inline theme/js/dataset-recommendations.js]: `_paq` should be an array so `const _paq = _paq || [];`

File: theme/js/dataset-recommendations.js [hunk 4]
```diff
@@ -20,6 +29,10 @@ function addRecos(recos) {
         recoChildContainer.appendChild(recoChildEmbed);
         recoContainer.appendChild(recoChildContainer);
     });
+
+    if (paq.push) {
```
Comment (noirbizarre) [inline theme/js/dataset-recommendations.js]: Not needed anymore if `_paq` is properly initialized as an array

File: udata_gouvfr/theme/templates/dataset/display.html [hunk 5]
```diff
@@ -2,9 +2,12 @@
 
 {% set community_subtitle = _('Explore with %(certifier)s', certifier=config.SITE_TITLE ) %}
 
+{% block extra_js %}
+<script src="{{ theme_static('dataset-recommendations.js') }}" type="text/javascript" onload="udataDatasetRecos.load()"></script>
```
Comment (noirbizarre) [inline udata_gouvfr/theme/templates/dataset/display.html]: Don't forget the `{{ super() }}` (avoid parent `extra_js` to be overwritten)

File: udata_gouvfr/theme/templates/dataset/display.html [hunk 6]
```diff
@@ -2,9 +2,12 @@
...
+{% endblock %}
+
 {% block left_column_bottom %}
 <!-- dataset recommendations -->
```
Comment (abulte) [inline udata_gouvfr/theme/templates/dataset/display.html]: Remove me

**Pull request merged**